Updated for Fix link to SWC spec #393

SridharJagannathan opened the pull request:
Updated SWC spec link, resolved check errors and errors associated with not installation of `XML` package

jefferis commented on tests/testthat/test-neuron-io.R:
```diff
@@ -875,8 +875,19 @@ test_that("we can write neuron to swc file",{
+  expect_equal(f<-write.neuron(y, dir=td, ext='.swc'),
                file.path(td,'EBH11R.swc'))
+  
+  expect_equal(f<-write.neuron(y, dir=td, format = 'swc', Force = TRUE),
```
This feels like a duplication.

jefferis commented on tests/testthat/test-neuron-io.R:
```diff
@@ -875,8 +875,19 @@ test_that("we can write neuron to swc file",{
+  swc_data <- read.delim(f, stringsAsFactors = FALSE)
+  
+  url_pattern <- "http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\\(\\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+"
+  swc_url <- stringr::str_extract(swc_data[[1]][1], url_pattern)
```
I think reading the first few lines and looking for `http` would be enough.

jefferis replied:
As mentioned I think adding stringr for this is a little overkill. I've had trouble compiling stringr on some linux platforms in the past. I might have gone for something like:

```
testurl="# See http://www.neuronland.org/NLMorphologyConverter/MorphologyFormats/SWC/Spec.html for more details"
sub(".*?(http.{3,}?)(\\s+.*|\\s*){0,1}$", "\\1", testurl)
```
but you can also use that url pattern.

jefferis commented on tests/testthat/test-neuron-io.R:
```diff
@@ -875,8 +875,19 @@ test_that("we can write neuron to swc file",{
+  expect_equal(f<-write.neuron(y, dir=td, format = 'swc', Force = TRUE),
+               file.path(td,'EBH11R.swc'))
+  swc_data <- read.delim(f, stringsAsFactors = FALSE)
+  
```
It took me a moment to realise why you were doing this. I think it's quite clever, but I wonder if it isn't overkill!

jefferis commented on tests/testthat/test-neuron-io.R:
```diff
@@ -875,8 +875,19 @@ test_that("we can write neuron to swc file",{
+  
+  #Check if the URL exists
+  expect_equal(TRUE,RCurl::url.exists(swc_url))
+  
```
1. There's an equivalent function in httr (which is already a direct dependency) called `url_ok`. However for reasons that are beyond me it has been deprecated. But you can redefined it as
```
url_ok<-function(x) identical(httr::status_code(httr::HEAD(x)), 200L)
expect_true(url_ok(swc_url))
```
2. You need to condition this test on whether the internet is available. You can do this by
```
if(nzchar(Sys.getenv("NAT_INTERNET_TESTS")))
```

SridharJagannathan commented:
Hi @jefferis 
I have updated your review comments:
1) Removed the pdf generated, can you please tell me why we don't have this auto generated pdf on the git?
2) I have updated the solution based on `baseR`.
3) Sure, would do separate commits from next time on.

jefferis commented:
> 1. Removed the pdf generated, can you please tell me why we don't have this auto generated pdf on the git?
It's produced when you run tests locally. It should be ignored as it's not required for the tests to run – it's a byproduct.
> 2. I have updated the solution based on baseR.
Great!
